core/jobs/types/base_validation_errors.py

```diff
@@ -22,20 +22,35 @@
 from core import utils
 from core.jobs import job_utils
 from core.jobs.types import job_run_result
+from core.jobs.types import model_property
+from core.platform import models
+
+from typing import Dict, Optional, Union
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
 
 
 class BaseAuditError(job_run_result.JobRunResult):
     """Base class for model audit errors."""
 
-    def __init__(self, message, model_or_kind, model_id=None):
+    def __init__(
+        self,
+        message: str,
+        model_or_kind: Union[base_models.BaseModel, str],
+        model_id: Optional[str] = None
+    ) -> None:
         """Initializes a new audit error.
 
         Args:
             message: str. The message describing the error.
-            model_or_kind: Model|bytes. If model_id is not provided, then this
+            model_or_kind: Model|str. If model_id is not provided, then this
                 is a model (type: BaseModel).
-                Otherwise, this is a model's kind (type: bytes).
-            model_id: bytes|None. The model's ID, or None when model_or_kind is
+                Otherwise, this is a model's kind (type: str).
+            model_id: str|None. The model's ID, or None when model_or_kind is
```

Comment:
These should be str not bytes. because if we consider them as bytes and pass them to utils.quoted() then this function throws an error `Object of type bytes is not JSON serializable`. https://github.com/oppia/oppia/runs/5751946533?check_suite_focus=true#step:8:1654

Also, values returned by `get_model_id` and `get_model_kind` is also str not bytes. 
So, that's why i added str here.